pkg/argocd/argocd.go

```diff
@@ -37,19 +37,33 @@ func Apply(ctx context.Context, config *rest.Config, namespace, argoImage string
 	if err != nil {
 		return err
 	}
+
 	deployments, err := clientset.AppsV1().Deployments(namespace).List(metav1.ListOptions{
 		LabelSelector: "app.kubernetes.io/part-of=argocd",
 	})
 	if err != nil {
 		return err
 	}
-	expectedCount := 4
-	foundCount := len(deployments.Items)
-	if foundCount == expectedCount {
-		klog.Infof("Found %d of %d deployments", foundCount, expectedCount)
+	expectedDeploymentCount := 3
+	foundDeploymentCount := len(deployments.Items)
+	if foundDeploymentCount == expectedDeploymentCount {
+		klog.Infof("Found %d of %d deployments", foundDeploymentCount, expectedDeploymentCount)
```

Comment:
We need to change the logic a little here. Otherwise it will return as soon as it found 3 deployments and won't look for the StatefulSet.